kernel/src/cpu/efer.rs

```diff
@@ -36,9 +36,13 @@ pub fn write_efer(efer: EFERFlags) {
 pub fn efer_init() {
     let mut efer = read_efer();
 
-    if cpu_has_nx() {
-        efer.insert(EFERFlags::NXE);
+    if !cpu_has_nx() {
+        // All processors that are capable of virtualization will support
+        // no-execute table entries, so there is no reason to support any
+        // processor that does not enumerate NX capability.
+        panic!("CPU does not support NX");
     }
```

Comment:
Again, style suggestion, feel free to ignore:

```suggestion
    // All processors that are capable of virtualization will support
    // no-execute table entries, so there is no reason to support any
    // processor that does not enumerate NX capability.
    assert!(cpu_has_nx(), "CPU does not support NX");
```